### PR TITLE
Distance

### PR DESCRIPTION
**Pull request by raokrutarth**

### CODE REVIEW

**Review by mpace965 — APPROVED** [review]
Looks good to me, tested on my machine

**Review by mpace965 — CHANGES_REQUESTED** [review]
The dataset needs to be put in and accessed from static storage.

**Review by TomFanella4 — CHANGES_REQUESTED** [review]
Can we put the changes to the dataset on hold? Making the large files even larger is going to make pruning take longer.

**Review by TomFanella4 — APPROVED** [review]

**Comment by raokrutarth** [comment]
@mpace965 is static storage google drive? I believe I'm writing the generation code as if we use the files directly from disk and Tom will transition that code to the cloud file access and clean up any large files in data/

File: backend/Handlers.go [hunk 1]
```diff
@@ -79,9 +79,39 @@ func Test(w http.ResponseWriter, r *http.Request) {
...
+
+		for _, u := range cachedUsers {
+			u.Profile.FormattedName = u.Profile.FirstName + " " + u.Profile.LastName
+			u.Profile.ID = u.ID
+			umap[u.ID] = u
+		}
+
+		if err := users.Update(umap); err != nil {
+			fmt.Println(err)
+		}
```
Comment (mpace965) [inline backend/Handlers.go]: `seedUser` can be removed in its entirety

File: backend/data/model.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+// 	// Start to Train.
+// 	m.Train(f1)
+// 	// m.Train(f2)
+// 	f1.Close()
+// 	// f2.Close()
+
+// 	// Save word vectors to a text file.
+// 	m.Save(destinationFileName)
+
+// }
```
Comment (mpace965) [inline backend/data/model.go:176]: What is the purpose of this file?

**Review by mpace965 — APPROVED** [review]
I had a couple of questions, but otherwise looks good for now.

**Review by mpace965 — CHANGES_REQUESTED** [review]
This branch does not currently build. I get this when running `go build`:

```
# meetover/backend
./Handlers.go:90:37: undefined: cachedUsers
./Handlers.go:92:21: undefined: cachedUsers
./Handlers.go:110:35: unknown field 'Long' in struct literal of type Geolocation
./Handlers.go:110:50: unknown field 'Lat' in struct literal of type Geolocation
./Location.go:37:25: center.Lat undefined (type Geolocation has no field or method Lat)
./Location.go:37:37: center.Long undefined (type Geolocation has no field or method Long)
./Location.go:37:49: point.Lat undefined (type Geolocation has no field or method Lat)
./Location.go:37:60: point.Long undefined (type Geolocation has no field or method Long)
./UserManagement.go:252:2: undefined: cachedUsers
```

**Comment by raokrutarth** [comment]
My bad on the build issue. Was not aware of the geoloc field name change.

File: backend/UserManagement.go [hunk 3]
```diff
@@ -107,6 +107,9 @@ type User struct {
 	IsMatchedNow bool           `json:"isMatched"` // set directly from the mobile app
 }
 
+// cachedUsers - temp var for testing
+var cachedUsers []User
```
Comment (TomFanella4) [inline backend/UserManagement.go]: Why do we still need `cachedUsers`? I thought we decided we would get all users from Firebase?

File: backend/UserManagement.go [hunk 4]
```diff
@@ -235,3 +238,19 @@ func InitUser(p Profile, aTokenResp ATokenResponse) (bool, error) {
 	}
 	return userExists, nil
 }
+
+// LoadTestUsers gets test users generated from a random data generator
+func LoadTestUsers() {
```
Comment (TomFanella4) [inline backend/UserManagement.go]: Related to `cachedUsers` comment.

File: backend/main.go [hunk 5]
```diff
@@ -16,6 +16,9 @@ func main() {
 	// database & chat
 	InitializeFirebase()
 
+	// demo data
+	// LoadTestUsers()
```
Comment (TomFanella4) [inline backend/main.go]: Related to `cachedUsers` comment.

File: backend/Handlers.go [hunk 6]
```diff
@@ -79,9 +79,39 @@ func Test(w http.ResponseWriter, r *http.Request) {
 	tt := params["testType"]
 	if tt == "profile" {
 		json.NewEncoder(w).Encode(strings.Replace(sampleProfile, "\n", "", -1))
+	} else if tt == "seedUser" {
```
Comment (TomFanella4) [inline backend/Handlers.go]: Related to `cachedUsers` comment.

File: backend/Location.go [hunk 7]
```diff
@@ -24,3 +30,35 @@ func QueryLocation(coords string) (Address, error) {
...
+	la1 = lat1 * math.Pi / 180
+	lo1 = lon1 * math.Pi / 180
+	la2 = lat2 * math.Pi / 180
+	lo2 = lon2 * math.Pi / 180
+	r = 6378100 // Earth radius in METERS
+	h := hsin(la2-la1) + math.Cos(la1)*math.Cos(la2)*hsin(lo2-lo1)
+	return 2 * r * math.Asin(math.Sqrt(h))
+}
+
+func makeTimestamp() int64 {
```
Comment (TomFanella4) [inline backend/Location.go]: `makeTimestamp()` can be removed because the timestamp is created on the client.

File: backend/data/model.go (hunk 2 above)
Comment (TomFanella4) [inline backend/data/model.go:176]: I'm also confused as to why we need this file.

**Review by TomFanella4 — CHANGES_REQUESTED** [review]
Branch now builds. Left several comments that need to be addressed before merged.

**Review by TomFanella4 — APPROVED** [review]

**Pull request merged**